SO_properties.py

```diff
@@ -636,12 +638,116 @@ def Tgas(self):
             return None
         return (self.gas_temperatures * (self.gas_masses / self.Mgas)).sum()
 
+    @lazy_property
+    def Tgas_cy_weighted(self):
+        if self.Ngas == 0:
+            return None
+        return (
+            self.gas_temperatures * (self.gas_compY.value / self.gas_compY.sum().value)
+        ).sum()
+
+    @lazy_property
+    def Tgas_cy_weighted_no_agn(self):
+        if self.Ngas == 0:
+            return None
+        return (
+            self.gas_temperatures[self.gas_no_agn]
+            * (
+                self.gas_compY[self.gas_no_agn].value
+                / self.gas_compY[self.gas_no_agn].sum().value
+            )
+        ).sum()
+
+    @lazy_property
+    def Tgas_cy_weighted_core_excision(self):
+        if self.Ngas == 0:
+            return None
+        return (
+            self.gas_temperatures[self.gas_selection_core_excision]
```

Comment:
This doesn't look right. gas_selection_core_excision has one element for every gas particle in the region, including those outside the SO radius. But self.gas_temperatures only contains particles within the SO radius. numpy doesn't raise an exception if a boolean index array has the wrong number of elements.

Reply:
Is that true? from my understanding (and tests), in line 296-298 the radius cut is applied to self.radius, so this has only particles within the SO, similar to self.gas_temperatures

Reply:
Actually I think this is ok. I hadn't spotted that the radius array already has the particles outside the SO radius filtered, so gas_selection_core_excision has one element per particle within the SO radius.